backend/types.ts

```diff
@@ -22,6 +22,17 @@ export type RegisterUserDTO = Omit<CreateUserDTO, "role">;
 
 export type AuthDTO = Token & UserDTO;
 
+export type LessonDTO = {
```

Comment:
As a consistency thing (since our course APIs implementation should look pretty similar): we followed the `Entity` example more closely and declared the `CourseRequestDTO`/`CourseResponseDTO` as interfaces in the `services/interfaces/ICourseService`.
- thoughts on interfaces vs. types for DTO declarations?
- calling classes in `services/interfaces` as `ILessonService` vs. `lessonService`
- might make sense to specify that this is the `LessonResponseDTO`